languages/js/src/helpers.ts

```diff
@@ -148,22 +142,14 @@ function parseNextExternal(d: obj): QueryEvent {
  *
  * @internal
  */
-function parseExternalCall({
-  args,
-  kwargs,
-  attribute,
-  call_id: callId,
-  instance,
-}: obj): QueryEvent {
+function parseExternalCall(event: unknown): QueryEvent {
+  if (!isObj(event)) throw new Error();
+  const { args, kwargs, attribute, call_id: callId, instance } = event;
+  if (!isArrayOfPolarTerms(args)) throw new Error();
   if (kwargs) throw new KwargsError();
-  if (
-    !Number.isSafeInteger(callId) ||
-    !isPolarTerm(instance) ||
-    typeof attribute !== 'string' ||
-    (args !== undefined &&
-      (!Array.isArray(args) || args.some((a: unknown) => !isPolarTerm(a))))
-  )
-    throw new Error();
+  if (typeof attribute !== 'string') throw new Error();
+  if (!isSafeInteger(callId)) throw new Error();
+  if (!isPolarTerm(instance)) throw new Error();
```

Comment:
could cut down on repetition a little with something like
```javascript
type Pred<A> = (a: A) => boolean;
function validate<A>(x: A, ...ps: Pred<A>[]) {
  for (cons p of ps) if (!p(x)) throw new Error();
}

validate(x, isPolarTerm)
```

Reply:
Tried this out:

1. Unfortunately it's not currently possible to parameterize a function with variadic generic types (w/o this I think we'd have to write a separate `validateN()` function for each arity we wanted to support since each generic type corresponds to a different predicate and the individual `x is X` assertions are ultimately `&`-ed together to create something like `x is A & B & C`).
2. I don't think it's possible to do this (but would love to be wrong about this!) b/c it's conflating the check that `a is A` with the enforcement of `throw new Error()`. The reason `if (!pred(a)) throw new Error();` works is b/c `pred(a)` asserts that `a is A` and `throw new Error()` basically acts as a "Do Not Pass Go; Do Not Collect $200" bomb-out, so that line tells the TS compiler, "if we make it past this line, `a is A`," i.e., the type of `a` has been narrowed to `A`.
3. We could probably write something a bit less generic and then do our own `as X` assertion if all of the predicates pass instead of trying to rely on the TS compiler's `is X` narrowing.